Assets/MixedRealityToolkit/Inspectors/ServiceInspectors/HandJointServiceInspector.cs

```diff
@@ -0,0 +1,271 @@
+            ShowHandJointFoldout = EditorGUILayout.Foldout(ShowHandJointFoldout, "Visible Hand Joints");
+            SessionState.SetBool(ShowHandJointFoldoutKey, ShowHandJointFoldout);
+
+            if (ShowHandJointFoldout)
+            {
+                #region setting buttons
+
+                EditorGUILayout.BeginHorizontal();
+
+                if (GUILayout.Button("All"))
+                {
+                    foreach (TrackedHandJoint joint in Enum.GetValues(typeof(TrackedHandJoint)))
+                    {
+                        if (joint == TrackedHandJoint.None)
+                        {
+                            continue;
+                        }
+
+                        SessionState.SetBool(showHandJointSettingKeys[joint], true);
+                        showHandJointSettings[joint] = true;
+                    }
+                }
+
+                if (GUILayout.Button("Fingers"))
+                {
+                    foreach (TrackedHandJoint joint in Enum.GetValues(typeof(TrackedHandJoint)))
+                    {
+                        bool setting = false;
+                        switch (joint)
+                        {
+                            case TrackedHandJoint.IndexTip:
+                            case TrackedHandJoint.IndexDistalJoint:
+                            case TrackedHandJoint.IndexKnuckle:
+                            case TrackedHandJoint.IndexMetacarpal:
+                            case TrackedHandJoint.IndexMiddleJoint:
+
+                            case TrackedHandJoint.MiddleTip:
+                            case TrackedHandJoint.MiddleDistalJoint:
+                            case TrackedHandJoint.MiddleKnuckle:
+                            case TrackedHandJoint.MiddleMetacarpal:
+                            case TrackedHandJoint.MiddleMiddleJoint:
+
+                            case TrackedHandJoint.PinkyTip:
+                            case TrackedHandJoint.PinkyDistalJoint:
+                            case TrackedHandJoint.PinkyKnuckle:
+                            case TrackedHandJoint.PinkyMetacarpal:
+                            case TrackedHandJoint.PinkyMiddleJoint:
+
+                            case TrackedHandJoint.RingTip:
+                            case TrackedHandJoint.RingDistalJoint:
+                            case TrackedHandJoint.RingKnuckle:
+                            case TrackedHandJoint.RingMetacarpal:
+                            case TrackedHandJoint.RingMiddleJoint:
+
+                            case TrackedHandJoint.ThumbTip:
+                            case TrackedHandJoint.ThumbDistalJoint:
+                            case TrackedHandJoint.ThumbMetacarpalJoint:
+                            case TrackedHandJoint.ThumbProximalJoint:
+                                setting = true;
+                                break;
+
+                            default:
+                                break;
+
```

Comment:
Nit: It's a bit weird reading the default case show up before the non-default case, if only because default always executes last regardless of the order here (so seeing out of code visually laid out out of the order in which it executes is a bit odd to see)

That being said, I can't find any coding guidelines that state this explicitly - it seems like it's implicit in guidelines:

[https://google.github.io/styleguide/javaguide.html#s4.8.4-switch](https://google.github.io/styleguide/javaguide.html#s4.8.4-switch)
[https://google.github.io/styleguide/cppguide.html#Loops_and_Switch_Statements](https://google.github.io/styleguide/cppguide.html#Loops_and_Switch_Statements)
[https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#es70-prefer-a-switch-statement-to-an-if-statement-when-there-is-a-choice](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#es70-prefer-a-switch-statement-to-an-if-statement-when-there-is-a-choice)

This comment is for here and below as well